src/parser/jobs/drk/modules/Darkside.tsx

```diff
@@ -13,28 +17,25 @@ const DARKSIDE_EXTENSION = {
 }
 const INITIAL_APPLICATION_FORGIVENESS = 2500
 
-export default class Darkside extends Module {
+export class Darkside extends Module {
 	static handle = 'Darkside'
-	static dependencies = [
-		'checklist',
-		'death',
-	]
 
 	static title = t('drk.darkside.title')`Darkside`
+	@dependency private checklist!: Checklist
+	@dependency private death!: Death
 
-	_currentDuration = 0
-	_downtime = 0
-	_lastEventTime = null
+	private _currentDuration = 0
+	private _downtime = 0
+	private _lastEventTime: number | null = null
 
-	constructor(...args) {
-		super(...args)
+	protected init() {
 		this.addEventHook('normaliseddamage', {by: 'player', abilityId: Object.keys(DARKSIDE_EXTENSION).map(Number)}, this._updateDarkside)
 		this.addEventHook('death', {to: 'player'}, this._onDeath)
 		this.addEventHook('raise', {to: 'player'}, this._onRaise)
 		this.addEventHook('complete', this._onComplete)
 	}
 
-	_updateDarkside(event) {
+	_updateDarkside(event: Event | NormalisedDamageEvent) {
```

Comment:
Would this not always be a `NormalisedDamageEvent`? Quick look at the type def has `DamageEvent | NormalisedDamageEvent` anyway but idk, hooks are black magic.

Reply:
I’m overloading _updateDarksideEvent both as the receiver for the event hook and as a function called from onDeath. If it gets called from onDeath, it gets a death event passed in. 

I should probably refactor a bit so both of the event receiver functions call a third function instead :facepalm:

Reply:
Makes sense, bit iffy to read but checks out so yolo.